CMakeLists.txt

```
@@ -33,11 +33,28 @@ if (${VARA_FEATURE_COLORED_OUTPUT})
   endif ()
 endif ()
 
+option(VARA_FEATURE_SANITIZE "Build vara-features with ASan and UBSan" FALSE)
+if (${VARA_FEATURE_SANITIZE})
+  set(CMAKE_CXX_FLAGS "-fsanitize=address,undefined ${CMAKE_CXX_FLAGS}")
+else ()
+  set(CMAKE_CXX_FLAGS "-fno-sanitize=address,undefined ${CMAKE_CXX_FLAGS}")
+endif ()
+
 option(VARA_FEATURE_BUILD_PYTHON_BINDINGS "Build vara-features python bindings" TRUE)
 if (${VARA_FEATURE_BUILD_PYTHON_BINDINGS})
+  if (${VARA_FEATURE_SANITIZE})
+    message(FATAL_ERROR "Cannot build python bindings together with sanitizers.")
+  endif ()
   add_compile_options(-fPIC)
 endif ()
 
+# Extra debug warning flags
+if (CMAKE_BUILD_TYPE EQUAL "Debug")
+  add_compile_options(-Wall -Wextra -Wno-unused-parameter -O0)
```

```suggestion
  add_compile_options(-Wall -Wextra -Wno-unused-parameter)
```
We should not force O0 here, llvm normal debug ("dev") builds use for example `O2`